### DISCUSSION TITLE
Rewrites can't be configured when using Vite SPA + Nx

### DISCUSSION THREAD

**Discussion by andyjessop** [vercel/community#2074] [Help]
We have an Nx monorepo setup running a Vite SPA app. The advice [here](https://vercel.com/docs/frameworks/vite#using-vite-to-make-spas), when using Vite SPA is to add a `vercel.json` to the root with the rewrite config, however, the root directory is not the root of the Vite project, so any `vercel.json` at the root would be applied universally for other apps in the repo.

Is there a way to add the `rewrites` config without a root-level `vercel.json`?

**Comment by andefred** [comment]
I have the same issue. Did you end up finding a solution to this?

**Reply by andyjessop** [reply]
Unfortunately  the solution was to not use Vercel. Luckily it was an easy decision because we were unhappy with it for other reasons too. So we didn't explore the solution fully.

**Comment by aiyeola** [comment]
Hi @andyjessop @andefred, I had this issue and have been able to figure it out.

the solution is in how you configure the project on Vercel. the `vercel.json` file should be in the root of the vite app.

set the root directory

<img width="833" alt="image" src="https://github.com/vercel/community/assets/47264345/48ebff01-2b67-4be0-9ea0-7a3688c438e0">

override the commands like so

<img width="833" alt="image" src="https://github.com/vercel/community/assets/47264345/3a77ac3d-bc1f-4ac1-9974-cecb0cb3d9ec">

for the output directory, depends on how you set up the outputPath for your app, however, point the command to wherever the build is.

**Reply by aiyeola** [reply]
I also ran into an issue with the **Ignored Build Step** command

the `npx nx-ignore vite-app` could not work because the root directory has changed. my workaround was to use custom bash script to do this check

<img width="866" alt="image" src="https://github.com/vercel/community/assets/47264345/084b18a2-e6c2-4dab-8514-b0710c392374">


`apps/vite-app/build.sh`

```
#!/bin/bash

echo "VERCEL_GIT_COMMIT_REF: $VERCEL_GIT_COMMIT_REF"

if git diff --name-only HEAD^ HEAD | grep "apps/vite-app/"; then
  # Proceed with the build
    echo "✅ Changes detected in apps/vite-app. Proceeding with the build."
  exit 1;

else
  # Don't build
  echo "🛑 - Build cancelled"
  exit 0;
fi
```

**Reply by Cacaci** [reply]
Thanks, it save me.
